merge fcrisciani/ip-util-check:fix-peers-list and adamancini/ip-util-check:swarmctl

adamancini opened the pull request.

adamancini committed:
add debug function to set -x

adamancini committed:
disable unloading client bundle

adamancini committed:
use swarmctl to get subnet size due to engine regression

adamancini committed:
Update README.md

Commit:
Fix use of Peer list

- If peer list is not available, consider 25% of the cluster using the
network and adjust the output message
- retab the file

Signed-off-by: Flavio Crisciani <flavio.crisciani@docker.com>

adamancini committed:
grammar edit

adamancini committed:
fix merge conflict: merge fix-peers branch

fcrisciani commented on README.md:
````diff
@@ -20,14 +20,29 @@ The script flags several potential conditions for each overlay:
-docker image build -t docker/ip-util-check .
-
+```
+docker build -t support/ip-util-check .
````
```suggestion
docker build -t docker/ip-util-check .
```

fcrisciani commented on README.md:
```diff
@@ -20,14 +20,29 @@ The script flags several potential conditions for each overlay:
+docker run -it --rm \
+ -v /var/run/docker.sock:/var/run/docker.sock \
+ -v /var/run/docker/swarm/control.sock:/var/run/swarmd.sock \
+ support/ip-util-check
```
```suggestion
 docker/ip-util-check
```

fcrisciani commented on Dockerfile:
```diff
@@ -1,4 +1,16 @@
-FROM alpine:latest
-RUN apk add --update bash docker jq
+FROM golang:1.11.0-alpine
```
```suggestion
FROM golang:1.11-alpine
```
At least we get the latest of the 1.11 train

fcrisciani commented on Dockerfile:
```diff
@@ -1,4 +1,16 @@
+  && unzip -d /usr/local protoc.zip include/\* bin/\* \
+  && rm -f protoc.zip
+WORKDIR /go/src/github.com/docker
+RUN git clone https://github.com/docker/swarmkit.git swarmkit && cd swarmkit && make bin/swarmctl && cp bin/swarmctl /usr/bin/swarmctl && rm -rf /go/src/github.com/docker/swarmkit
```
you may want to use multi stage build so that the compilation steps stays in the builder image and only the final binaries are copied to the tool image

adamancini replied:
@fcrisciani thanks, that's the next thing I'm working on.  Originally I had compiled swarmctl in a first stage, but had trouble getting the swarmctl binary copied & working in the second stage, so I combined them so I could get it "just working"

adamancini replied:
resolved - updated with multistage

fcrisciani commented:
you may also want to squash all the commits into 1, so that the history is clean

adamancini commented:
@fcrisciani thanks for the suggestions

Do you know if we publish a `swarmkit` image via CI/CD such that I might be able to COPY --from=docker/swarmkit to pull that binary?

fcrisciani commented:
@adamancini I doubt, but maybe you can ask to the release team if we have such an image somewhere.

adamancini committed:
use 'swarmctl' for 17.06 and estimate peers when we don't get ipam info

merge fcrisciani/ip-util-check:fix-peers-list and adamancini/ip-util-check:swarmctl

clay-moss committed:
Update Dockerfile

- golang version 1.18.1
- alpine 3.15
- docker client 20.10.10
- protobuf 3.20.0

clay-moss committed:
Merge pull request #4 from Mirantis/cmoss-version-updates

version updates

clay-moss committed:
include ingress network

include the ingress overlay network in the IP utilization calculations.

clay-moss committed:
Merge pull request #5 from Mirantis/cmoss-include-ingress

include ingress network
